<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>713</width>
    <height>551</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MTG Editor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" rowspan="4" colspan="6">
     <widget class="GLMTGEditor" name="mtgeditor" native="true"/>
    </item>
    <item row="0" column="6">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>Clip</string>
      </property>
     </widget>
    </item>
    <item row="1" column="6">
     <widget class="QCheckBox" name="visibilityEnabled">
      <property name="text">
       <string/>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="2" column="6">
     <widget class="QSlider" name="backVisibilitySlider">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
      <property name="invertedControls">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="3" column="6">
     <widget class="QSlider" name="frontVisibilitySlider">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>PointSize</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QSlider" name="pointSizeSlider">
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>10</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="4" column="2">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>PointFilter</string>
      </property>
     </widget>
    </item>
    <item row="4" column="3">
     <widget class="QSlider" name="pointFilterSlider">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>4</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximum">
       <number>1000</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="4" column="4">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>NodeSize</string>
      </property>
     </widget>
    </item>
    <item row="4" column="5" colspan="2">
     <widget class="QSlider" name="nodeSizeSlider">
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>10</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>713</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuExport_MTG">
     <property name="title">
      <string>Export MTG</string>
     </property>
     <addaction name="actionExportNodeList"/>
    </widget>
    <widget class="QMenu" name="menuExport_View">
     <property name="title">
      <string>Export View</string>
     </property>
     <addaction name="actionExportGeom"/>
     <addaction name="actionSaveSnapshot"/>
    </widget>
    <widget class="QMenu" name="menuExport_Points">
     <property name="title">
      <string>Export Points</string>
     </property>
     <addaction name="actionExportPoints"/>
     <addaction name="actionExportContractedPoints"/>
    </widget>
    <addaction name="actionOpen_Project"/>
    <addaction name="actionSave_Project"/>
    <addaction name="separator"/>
    <addaction name="actionOpenMTG"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveMTG"/>
    <addaction name="separator"/>
    <addaction name="actionImportPoints"/>
    <addaction name="actionImportContractedPoints"/>
    <addaction name="separator"/>
    <addaction name="menuExport_MTG"/>
    <addaction name="menuExport_View"/>
    <addaction name="menuExport_Points"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionViewMTG"/>
    <addaction name="actionViewControlPoints"/>
    <addaction name="actionViewPoints"/>
    <addaction name="actionViewContractedPoints"/>
    <addaction name="actionView3DModel"/>
    <addaction name="separator"/>
    <addaction name="actionRefreshView"/>
    <addaction name="separator"/>
    <addaction name="actionAdjustView"/>
   </widget>
   <widget class="QMenu" name="menuProcess">
    <property name="title">
     <string>Process</string>
    </property>
    <widget class="QMenu" name="menuRadius">
     <property name="title">
      <string>Radius</string>
     </property>
     <addaction name="actionComputeRadius"/>
     <addaction name="separator"/>
     <addaction name="actionAverageRadius"/>
     <addaction name="actionFilterRadius"/>
    </widget>
    <widget class="QMenu" name="menuAdd_Root">
     <property name="title">
      <string>Add Root</string>
     </property>
     <addaction name="actionRootBottom"/>
     <addaction name="actionRootTop"/>
    </widget>
    <widget class="QMenu" name="menuReconstruction">
     <property name="title">
      <string>Reconstruction</string>
     </property>
     <addaction name="actionXuReconstruction"/>
     <addaction name="actionCreateSkeleton"/>
    </widget>
    <widget class="QMenu" name="menuContraction">
     <property name="title">
      <string>Contraction</string>
     </property>
     <addaction name="actionEuclidianContraction"/>
     <addaction name="actionRiemannianContraction"/>
     <addaction name="actionContract"/>
    </widget>
    <addaction name="menuAdd_Root"/>
    <addaction name="separator"/>
    <addaction name="actionSubSampling"/>
    <addaction name="separator"/>
    <addaction name="menuContraction"/>
    <addaction name="separator"/>
    <addaction name="menuReconstruction"/>
    <addaction name="separator"/>
    <addaction name="menuRadius"/>
    <addaction name="separator"/>
    <addaction name="actionAngleEstimation"/>
    <addaction name="separator"/>
    <addaction name="actionCheckMTG"/>
    <addaction name="separator"/>
    <addaction name="actionTagScale"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="actionRevolveAroundScene"/>
    <addaction name="actionShowAll"/>
    <addaction name="separator"/>
    <addaction name="actionReorient"/>
   </widget>
   <widget class="QMenu" name="menuLoad">
    <property name="title">
     <string>Load</string>
    </property>
    <addaction name="actionPuu1"/>
    <addaction name="actionPuu3"/>
    <addaction name="actionCherry"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuView"/>
   <addaction name="menuProcess"/>
   <addaction name="menuLoad"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNodesInfo">
   <property name="text">
    <string>Import Debug Info</string>
   </property>
  </action>
  <action name="actionImportPoints">
   <property name="text">
    <string>Import Points</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionOpenMTG">
   <property name="text">
    <string>Open MTG</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSaveMTG">
   <property name="text">
    <string>Save MTG as</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionViewMTG">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>MTG</string>
   </property>
  </action>
  <action name="actionViewPoints">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Points</string>
   </property>
  </action>
  <action name="actionAdjustView">
   <property name="text">
    <string>Adjust View</string>
   </property>
  </action>
  <action name="actionViewControlPoints">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Control Points</string>
   </property>
  </action>
  <action name="actionRefreshView">
   <property name="text">
    <string>Refresh View</string>
   </property>
  </action>
  <action name="actionContract">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Preuksakarn et al.</string>
   </property>
  </action>
  <action name="actionCreateSkeleton">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Preuksakarn et al.</string>
   </property>
  </action>
  <action name="actionExportGeom">
   <property name="text">
    <string>As PlantGL file</string>
   </property>
  </action>
  <action name="actionViewContractedPoints">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Contracted Points</string>
   </property>
  </action>
  <action name="actionImportContractedPoints">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Import Contracted Points</string>
   </property>
  </action>
  <action name="actionExportContractedPoints">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Export Contracted Points</string>
   </property>
  </action>
  <action name="actionView3DModel">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>3D Model</string>
   </property>
  </action>
  <action name="actionComputeRadius">
   <property name="text">
    <string>Computing</string>
   </property>
  </action>
  <action name="actionAverageRadius">
   <property name="text">
    <string>Averaging</string>
   </property>
  </action>
  <action name="actionFilterRadius">
   <property name="text">
    <string>Filtering</string>
   </property>
  </action>
  <action name="actionCheckMTG">
   <property name="text">
    <string>CheckMTG</string>
   </property>
  </action>
  <action name="actionExportNodeList">
   <property name="text">
    <string>As Node List</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionPuu1">
   <property name="text">
    <string>puu1</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+1</string>
   </property>
  </action>
  <action name="actionPuu3">
   <property name="text">
    <string>puu3</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+3</string>
   </property>
  </action>
  <action name="actionCherry">
   <property name="text">
    <string>cherry</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+9</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save MTG</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveSnapshot">
   <property name="text">
    <string>As snapshot</string>
   </property>
  </action>
  <action name="actionRevolveAroundScene">
   <property name="text">
    <string>Revolve Around Scene</string>
   </property>
  </action>
  <action name="actionShowAll">
   <property name="text">
    <string>Show All</string>
   </property>
  </action>
  <action name="actionReorient">
   <property name="text">
    <string>Swap Y and Z</string>
   </property>
  </action>
  <action name="actionXuReconstruction">
   <property name="text">
    <string>Xu et al.</string>
   </property>
  </action>
  <action name="actionRootBottom">
   <property name="text">
    <string>Bottom</string>
   </property>
  </action>
  <action name="actionRootTop">
   <property name="text">
    <string>Top</string>
   </property>
  </action>
  <action name="actionSubSampling">
   <property name="text">
    <string>Sub-Sampling</string>
   </property>
  </action>
  <action name="actionTest">
   <property name="text">
    <string>test</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="actionEuclidianContraction">
   <property name="text">
    <string>Euclidian Contraction</string>
   </property>
  </action>
  <action name="actionRiemannianContraction">
   <property name="text">
    <string>Riemannian Contraction</string>
   </property>
  </action>
  <action name="actionAngleEstimation">
   <property name="text">
    <string>Angle Estimation</string>
   </property>
  </action>
  <action name="actionAs_nothing">
   <property name="text">
    <string>As nothing</string>
   </property>
  </action>
  <action name="actionAs_picture">
   <property name="text">
    <string>As picture</string>
   </property>
  </action>
  <action name="actionExportPoints">
   <property name="text">
    <string>Export Current Points</string>
   </property>
  </action>
  <action name="actionOpen_Project">
   <property name="text">
    <string>Open Project</string>
   </property>
  </action>
  <action name="actionSave_Project">
   <property name="text">
    <string>Save Project</string>
   </property>
  </action>
  <action name="actionTagScale">
   <property name="text">
    <string>Tag Scale</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLMTGEditor</class>
   <extends>QWidget</extends>
   <header>mtgeditorwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>visibilityEnabled</sender>
   <signal>toggled(bool)</signal>
   <receiver>frontVisibilitySlider</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>694</x>
     <y>36</y>
    </hint>
    <hint type="destinationlabel">
     <x>694</x>
     <y>405</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>visibilityEnabled</sender>
   <signal>toggled(bool)</signal>
   <receiver>backVisibilitySlider</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>694</x>
     <y>36</y>
    </hint>
    <hint type="destinationlabel">
     <x>694</x>
     <y>165</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
